Mathlib/Topology/Algebra/InfiniteSum/Group.lean

```diff
@@ -397,3 +402,51 @@ theorem tprod_const [T2Space G] (a : G) : ∏' _ : β, a = a ^ (Nat.card β) :=
       simpa [multipliable_const_iff] using ha
 
 end IsTopologicalGroup
+
+section CommGroupWithZero
+variable {K : Type*} [CommGroupWithZero K] [TopologicalSpace K] [ContinuousMul K] {f g : α → K}
+/-!
+## Groups with a zero
+
+These lemmas apply to a `CommGroupWithZero`; the most familiar case is when `K` is a field. These
+are specific to the product setting and do not have a sensible additive analogue.
+-/
+
+open Finset in
+lemma HasProd.congr_cofinite₀ {c : K} (hc : HasProd f c) {s : Finset α}
+    (hs : ∀ a ∈ s, f a ≠ 0) (hs' : ∀ a ∉ s, f a = g a) :
+    HasProd g (c * ((∏ i ∈ s, g i) / ∏ i ∈ s, f i)) := by
+  classical
+  refine (Tendsto.mul_const ((∏ i ∈ s, g i) / ∏ i ∈ s, f i) hc).congr' ?_
+  filter_upwards [eventually_ge_atTop s] with t ht
+  calc (∏ i ∈ t, f i) * ((∏ i ∈ s, g i) / ∏ i ∈ s, f i)
+  _ = ((∏ i ∈ s, f i) * ∏ i ∈ t \ s, g i) * _ := by
+    conv_lhs => rw [← union_sdiff_of_subset ht, prod_union disjoint_sdiff,
+      prod_congr rfl fun i hi ↦ hs' i (mem_sdiff.mp hi).2]
+  _ = (∏ i ∈ s, g i) * ∏ i ∈ t \ s, g i := by
+    rw [← mul_div_assoc, ← div_mul_eq_mul_div, ← div_mul_eq_mul_div, div_self, one_mul, mul_comm]
+    exact prod_ne_zero_iff.mpr hs
+  _ = ∏ i ∈ t, g i := by
+    rw [← prod_union disjoint_sdiff, union_sdiff_of_subset ht]
+
+lemma tsum_congr_cofinite₀ [T2Space K] (hc : Multipliable f) {s : Finset α}
+    (hs : ∀ a ∈ s, f a ≠ 0) (hs' : ∀ a ∉ s, f a = g a) :
+    ∏' i, g i =  ((∏' i, f i) * ((∏ i ∈ s, g i) / ∏ i ∈ s, f i)) := by
```

Comment:
It could be nice to have a "multiplied out" version, ie where both sides are of the form `_ * _`, but this is optional

Reply:
I'm not sure it would be useful, so let's agree to disagree on that.